src/proxy-ui-api/frontend/src/views/Clients/ServiceClients/ServiceClientAccessRights.vue

```diff
@@ -189,23 +221,42 @@ export default Vue.extend({
       this.showConfirmDeleteAll = false;
 
       api
-        .post(`/clients/${this.id}/service-clients/${this.serviceClientId}/access-rights/delete`,
-          {items: this.serviceClientAccessRights.map((item: AccessRight) => ({service_code: item.service_code}))})
+        .post(
+          `/clients/${this.id}/service-clients/${this.serviceClientId}/access-rights/delete`,
+          {
+            items: this.serviceClientAccessRights.map((item: AccessRight) => ({
+              service_code: item.service_code,
+            })),
+          },
+        )
         .then(() => {
           this.$store.dispatch('showSuccess', 'serviceClients.removeSuccess');
           this.serviceClientAccessRights = [];
         })
         .catch((error: any) => this.$store.dispatch('showError', error));
-
     },
     serviceCandidates(): ServiceCandidate[] {
-      return serviceCandidatesForServiceClient(this.clientServiceDescriptions, this.serviceClientAccessRights);
+      return serviceCandidatesForServiceClient(
+        this.clientServiceDescriptions,
+        this.serviceClientAccessRights,
+      );
+    },
+
+    keyedServiceClientAccessRights() {
+      let index = 0;
+      const arrWithKeys = [];
+      for (const accessRight of this.serviceClientAccessRights) {
+        arrWithKeys[index] = accessRight as UiAccessRight;
+        arrWithKeys[index].uiKey = index;
+        index++;
+      }
+
+      return arrWithKeys;
```

Comment:
i think this could be simplified a bit with just a map-function something like this:

`
keyedServiceClientAccessRights(): UiAccessRight[] {
      return this.serviceClientAccessRights.map( (sca: AccessRight, index: number) => {
        return { ... sca, uiKey: index };
      }) as UiAccessRight[];
    },
`

Reply:
fixed